cltk/lemmatize/latin/backoff.py

```diff
@@ -30,433 +32,257 @@
-    """"""
-    def __init__(self, backoff=None):
-        """Setup for SequentialBackoffLemmatizer()
 
-        :param backoff: Next lemmatizer in backoff chain.
+    def tag_one(self, tokens, index, history):
+        """
+        Determine an appropriate tag for the specified token, and
+        return that tag.  If this tagger is unable to determine a tag
+        for the specified token, then its backoff tagger is consulted.
+
+        :rtype: tuple
+        :type tokens: list
+        :param tokens: The list of words that are being tagged.
+        :type index: int
+        :param index: The index of the word whose tag should be
+            returned.
+        :type history: list(str)
+        :param history: A list of the tags for all words before index.
         """
-        LemmatizerI.__init__(self)
-        SequentialBackoffTagger.__init__(self, backoff)
+        lemma = None
+        for tagger in self._taggers:
+            lemma = tagger.choose_tag(tokens, index, history)
+            if lemma is not None:
+                break
+        return lemma, tagger
 
-    def lemmatize(self, tokens):
-        """Transform tag method into custom method for lemmatizing tasks. Can
-        be overwritten by specific instances where list of tokens should
-        be handled in a different manner. (Cf. IdentityLemmatizer)
 
-        :param tokens: List of tokens to be lemmatized
-        :return: Tuple of the form (TOKEN, LEMMA)
+    def lemmatize(self, tokens):
         """
-        return SequentialBackoffLemmatizer.tag(self, tokens)
+        Transform tag method into custom method for lemmatizing
+        tasks. Cf. ``tag`` method above.
+        """
+        return self.tag(tokens)
 
-    def choose_tag(self, tokens, index, history):
-        """Override choose_tag with lemmatizer-specific method for various
-        methods that expect a method with this name.
 
-        :param tokens: List of tokens to be lemmatized
-        :param index: Int with current token
-        :param history: List with tokens that have already been lemmatized
-        :return: String with lemma, if found; otherwise NONE
-        """
-        return self.choose_lemma(tokens, index, history)
+class DefaultLemmatizer(SequentialBackoffLemmatizer):
+    """
+    Lemmatizer that assigns the same lemma to every token.
 
+        >>> from cltk.lemmatize.latin.backoff import DefaultLemmatizer
+        >>> default_lemmatizer = DefaultLemmatizer('UNK')
+        >>> list(default_lemmatizer.lemmatize('arma virumque cano'.split()))
+        [('arma', 'UNK'), ('virumque', 'UNK'), ('cano', 'UNK')]
 
-class DefaultLemmatizer(SequentialBackoffLemmatizer, DefaultTagger):
-    """"""
-    def __init__(self, lemma=None):
-        """Setup for DefaultLemmatizer().
+    Useful as the final tagger in a chain, e.g. to assign 'UNK' to all
+    remaining unlemmatized tokens.
+
+    :type lemma: str
+    :param lemma: Lemma to assign to each token
+    """
```

Comment:
Minor nudge: probably make the doctest example last in the function, it makes it a little easier to read